exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/ColumnType.kt

```diff
@@ -139,6 +139,7 @@ class IntegerColumnType : ColumnType() {
     override fun valueFromDB(value: Any): Any = when(value) {
         is Int -> value
         is Number -> value.toInt()
+        is String -> value.toIntOrNull() ?: value
```

Comment:
Some functions returns string value even when applied on IntegerColumn. So extention is to support that and convert the value to integer if possible. Otherwise return it as a string.

Reply:
I would prefer to fail at this point with the meaningful exception instead of returning String from integer column/expression as it will fail with `ClassCastException` later.